src/shared/Core/Settings.cs

```diff
@@ -334,6 +336,28 @@ public IEnumerable<string> GetSettingValues(string envarName, string section, st
             {
                 IGitConfiguration config = _git.GetConfiguration();
 
+                // Enumerate all configuration entries with the correct section and property name
+                // and make a local copy of them here to avoid needing to call `TryGetValue` on the
+                // IGitConfiguration object multiple times in a loop below.
+
+                // This is a performance optimisation to avoid calling `TryGet` on the IGitConfiguration
+                // object multiple times in a loop below, or each time this entire method is called.
+                // The assumption is that the configuration entries will not change between invocations
```

Comment:
```suggestion
                // The assumption is that the configuration entries will not during a single invocation
```

If I understand correctly, this caching is scoped to a single Git Credential Manager process (i.e., a single invocation). "Will not change between invocations" to me implies an expectation that a Git config won't change between two separate calls to GCM, which isn't a guarantee.